allennlp/training/metric_tracker.py

```diff
@@ -120,3 +123,12 @@ def should_stop_early(self) -> bool:
             return False
         else:
             return self._epochs_with_no_improvement >= self._patience
+
+    # def best_epoch_metrics(self):
+    #     return self._best_epoch_metrics
+
+    # def update_best_epoch_metrics(self, metrics: Dict[str, Float]):
+    #     self._best_epoch_metrics = metrics
+
+    def best_epoch(self, last_run_epoch_num: int):
+        return last_run_epoch_num - self._epochs_with_no_improvement
```

Comment:
I find this method surprising and unintuitive, why not just store `best_epoch` explicitly?

Reply:
Updated 👍